test/ATokenCheckpointed.t.sol

```diff
@@ -0,0 +1,1974 @@
+      if (_supportTypeA == uint8(VoteType.Against)) assertEq(_againstVotes, _weight - _transferAmount);
+      if (_supportTypeA == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _weight - _transferAmount);
+      if (_supportTypeB == uint8(VoteType.For)) assertEq(_forVotes, _transferAmount);
+      if (_supportTypeB == uint8(VoteType.Against)) assertEq(_againstVotes, _transferAmount);
+      if (_supportTypeB == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _transferAmount);
+      // forgefmt: disable-end
+    }
+  }
+}
+
+contract GetPastStoredBalanceTest is AaveAtokenForkTest {
+  function test_GetPastStoredBalanceCorrectlyReadsCheckpoints() public {
+    _initiateRebasing();
+
+    address _who = address(0xBEEF);
+    uint256 _amountA = 42 ether;
+    uint256 _amountB = 3 ether;
+
+    uint256[] memory _rawBalances = new uint256[](3);
+
+    // Deposit.
+    _mintGovAndSupplyToAave(_who, _amountA);
+    _rawBalances[0] = aToken.exposedRawBalanceOf(_who);
+
+    // It's important that this be greater than a ray, since Aave uses this
+    // index when determining the raw stored balance. If it were a ray, the
+    // stored balance would just equal the supplied amount and this test would
+    // be less meaningful.
+    assertGt(
+      pool.getReserveData(address(govToken)).liquidityIndex,
+      1e27,
+      "liquidityIndex has not changed, is rebasing occuring?"
+    );
+
+    // The supplied amount should be less than the raw balance, which was
+    // scaled down by the reserve liquidity index.
+    assertLt(_rawBalances[0], _amountA, "supply wasn't reduced by liquidityIndex");
+
+    // Advance the clock.
+    uint256 _blocksJumped = 42;
+    vm.roll(block.number + _blocksJumped);
+    vm.warp(block.timestamp + 42 days);
+
+    // Rebasing should not affect the raw balance.
+    assertEq(
+      aToken.exposedRawBalanceOf(_who),
+      _rawBalances[0],
+      "raw balance shouldn't be affected by rebasing"
+    );
+
+    // getPastStoredBalance should match the initial raw balance.
+    assertEq(
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumped + 1),
+      _rawBalances[0],
+      "getPastStoredBalance does not match the initial raw balance"
+    );
+
+    // getPastStoredBalance should be able to give us the rebased balance at an
+    // intermediate point.
+    assertEq(
+      aToken.getPastStoredBalance(
+        _who,
+        block.number - (_blocksJumped / 3) // 1/3 is just an arbitrary point.
+      ),
+      _rawBalances[0]
+    );
+
+    // Deposit again to make things more complicated.
+    _mintGovAndSupplyToAave(_who, _amountB);
+    _rawBalances[1] = aToken.exposedRawBalanceOf(_who);
+
+    // Advance the clock.
+    uint256 _blocksJumpedSecondTime = 100;
+    vm.roll(block.number + _blocksJumpedSecondTime);
+    vm.warp(block.timestamp + 100 days);
+
+    // Rebasing should not affect the raw balance.
+    assertEq(aToken.exposedRawBalanceOf(_who), _rawBalances[1]);
```

Comment:
Similar no-op test that will always be true

Reply:
Removed in aa6ca4c